server/src/main/java/org/apache/kafka/server/config/AbstractKafkaConfig.java

```diff
@@ -79,7 +80,11 @@ public AbstractKafkaConfig(ConfigDef definition, Map<?, ?> originals, Map<String
     }
 
     public List<String> logDirs() {
-        return Csv.parseCsvList(Optional.ofNullable(getString(ServerLogConfigs.LOG_DIRS_CONFIG)).orElse(getString(ServerLogConfigs.LOG_DIR_CONFIG)));
+        return Optional.ofNullable(getList(ServerLogConfigs.LOG_DIRS_CONFIG))
+                .orElse(Arrays.stream(getString(ServerLogConfigs.LOG_DIR_CONFIG).split(",")).toList())
```

Comment:
Should we just define LOG_DIR_CONFIG as a List then? If so, we need to update the KIP.

Reply:
I think making LOG_DIR_CONFIG a LIST feels a bit odd. While it can be split by commas, if this configuration is allowed to have multiple values, it essentially overlaps with LOG_DIRS_CONFIG. If that’s the case, we might consider deprecating LOG_DIR_CONFIG and updating LOG_DIRS_CONFIG so that it does not allow null entries.

Reply:
Yes, I’ve updated this change in the PR.